pymbolic/geometric_algebra/mapper.py

```diff
@@ -324,11 +324,11 @@ def map_product(self, expr):
 
             result = new_result
 
-        from pymbolic.primitives import flattened_sum
+        from pymbolic.primitives import flattened_product, flattened_sum
         return flattened_sum([
-                    type(expr)(tuple([
+                    flattened_product([
```

Comment:
Could you add documentation to `flattened_product` to state that it will not reorder and therefore does not require products to be commutative?

Reply:
Added a little comment in 9a30523. How does that look?

`flattened_sum` and `flattened_product` aren't added to the docs though. Should they be? There's a nice "Helper Functions" section there where they would fit well enough.

Reply:
Added in 58e6069!